core/client/src/main/java/alluxio/client/file/UnderFileSystemFileInStream.java

```diff
@@ -57,12 +58,14 @@
    *
    * @param address worker address to read from
    * @param ufsFileId worker specific file id referencing the file to read
+   * @param pos position to read in bytes from the start of the file
    * @param reader a reader for reading from the worker
    */
-  public UnderFileSystemFileInStream(InetSocketAddress address, long ufsFileId,
+  public UnderFileSystemFileInStream(InetSocketAddress address, long ufsFileId, long pos,
       UnderFileSystemFileReader reader) {
     mAddress = address;
     mUfsFileId = ufsFileId;
+    mPos = pos;
```

Comment:
hmm. How does this work?  The reader won't start reading from mPos automatically. Maybe just keep it in the old way where we explicitly skip (or seek in your new code).